Sources/AuthFoundation/OAuth2/OAuth2ClientConfiguration.swift

```diff
@@ -0,0 +1,113 @@
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+extension OAuth2Client {
+    public enum PropertyListConfigurationError: Error {
+        case defaultPropertyListNotFound
+        case invalidPropertyList(url: URL)
+        case cannotParsePropertyList(_ error: Error?)
+        case missingConfigurationValues
+    }
+    
+    /// Utility struct used internally to process `Okta.plist` and other similar client configuration files.
+    ///
+    /// > Important: This struct is intended for internal use, and may be subject to change.
+    public struct PropertyListConfiguration {
+        public let issuer: URL
+        public let clientId: String
+        public let scopes: String
+        public let redirectUri: URL?
+        public let logoutRedirectUri: URL?
+        public let additionalParameters: [String: String]?
+
+        public init() throws {
+            guard let file = Bundle.main.url(forResource: "Okta", withExtension: "plist") else {
+                throw PropertyListConfigurationError.defaultPropertyListNotFound
+            }
+            
+            try self.init(plist: file)
+        }
+        
+        public init(plist fileURL: URL) throws {
+            guard fileURL.isFileURL else {
+                throw PropertyListConfigurationError.invalidPropertyList(url: fileURL)
+            }
+            
+            let plistContent: Any
+            do {
+                let data = try Data(contentsOf: fileURL)
+                plistContent = try PropertyListSerialization.propertyList(from: data,
+                                                                          options: [],
+                                                                          format: nil)
+            } catch {
+                throw PropertyListConfigurationError.cannotParsePropertyList(error)
+            }
+            
+            guard let dict = plistContent as? [String: String] else {
+                throw PropertyListConfigurationError.cannotParsePropertyList(nil)
+            }
+            
+            guard let clientId = dict["clientId"],
+                  !clientId.isEmpty,
+                  let issuer = dict["issuer"],
+                  let issuerUrl = URL(string: issuer),
+                  let scopes = dict["scopes"],
+                  !scopes.isEmpty
+            else {
+                throw PropertyListConfigurationError.missingConfigurationValues
+            }
+            
+            let redirectUri: URL?
+            if let redirectUriString = dict["redirectUri"] {
+                redirectUri = URL(string: redirectUriString)
+            } else {
+                redirectUri = nil
+            }
+            
+            let logoutRedirectUri: URL?
+            if let logoutRedirectUriString = dict["logoutRedirectUri"] {
```

Comment:
could use an enum for these keys, but I'm not against short strings (up to you)

Reply:
A subsequent PR later on will change the names of these keys to be more Apple-esque, but the older (e.g. these current) keys will still be needed for backward compatibility.  So having enum-based keys might make that challenging.